controllers/static/static_controller.go

```diff
@@ -0,0 +1,103 @@
+package static
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	flowslatest "github.com/netobserv/network-observability-operator/apis/flowcollector/v1beta2"
+	"github.com/netobserv/network-observability-operator/controllers/consoleplugin"
+	"github.com/netobserv/network-observability-operator/controllers/reconcilers"
+	"github.com/netobserv/network-observability-operator/pkg/helper"
+	"github.com/netobserv/network-observability-operator/pkg/manager"
+	"github.com/netobserv/network-observability-operator/pkg/manager/status"
+)
+
+const (
+	initReconcileAttempts = 5
+)
+
+type Reconciler struct {
+	client.Client
+	mgr    *manager.Manager
+	status status.Instance
+}
+
+func Start(ctx context.Context, mgr *manager.Manager) error {
+	log := log.FromContext(ctx)
+	log.Info("Starting Static controller")
+	r := Reconciler{
+		Client: mgr.Client,
+		mgr:    mgr,
+		status: mgr.Status.ForComponent(status.StaticPlugin),
+	}
+
+	// force reconcile at startup
+	go r.InitReconcile(ctx)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&flowslatest.FlowCollector{}, reconcilers.IgnoreStatusChange).
+		Named("staticPlugin").
+		Complete(&r)
+}
+
+func (r *Reconciler) InitReconcile(ctx context.Context) {
+	log := log.FromContext(ctx)
+	log.Info("Initializing resources...")
+
+	for attempt := range initReconcileAttempts {
+		// delay the reconcile calls to let some time to the cache to load
+		time.Sleep(5 * time.Second)
+		_, err := r.Reconcile(ctx, ctrl.Request{})
+		if err != nil {
+			log.Error(err, "Error while doing initial reconcile", "attempt", attempt)
+		} else {
+			return
+		}
+	}
+}
+
+// Reconcile is the controller entry point for reconciling current state with desired state.
+// It manages the controller status at a high level. Business logic is delegated into `reconcile`.
+func (r *Reconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result, error) {
+	l := log.Log.WithName("staticPlugin") // clear context (too noisy)
+	ctx = log.IntoContext(ctx, l)
+
+	r.status.SetUnknown()
+	defer r.status.Commit(ctx, r.Client)
+
```

Comment:
After setting "Unknown", I think this controller should return if openshift isn't detected, right?
We could check it in Kind

Reply:
That's done in the static reconciler using `HasConsolePlugin` function:
https://github.com/netobserv/network-observability-operator/blob/1080b69c915d647c11b75da7d39976b537171dbe/controllers/consoleplugin/consoleplugin_static_reconciler.go#L40-L77

If console plugin is not available, nothing is deployed and the status goes ready

Reply:
The static controller could deploy something else than the console plugin in future so I think it's better to keeps things separated here. WDYT ?

Reply:
ah ok, yes sounds good, thanks!